src/libasr/pass/replace_symbolic.cpp

```diff
@@ -54,6 +54,7 @@ class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisi
         ASR::Function_t &xx = const_cast<ASR::Function_t&>(x);
         SymbolTable* current_scope_copy = this->current_scope;
         this->current_scope = xx.m_symtab;
+        SymbolTable* module_scope = this->current_scope->parent;
```

Comment:
Clear `symbolic_vars` here, so that we will store only the symbolics that are present in the current scope.

Reply:
Or clear it at the end of the `visit_Function`.

Reply:
Ohh ok now I realize what the error  was . Thanks for pointing this out , I've tried to address this through the latest commit.